linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1800,24 +1849,22 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
             val importReplacement = ImportReplacement(importTarget, moduleName,
                 jsNativeLoadSpec.path, newSimpleState(Unused), cancelFun)
 
-            val (newCaptureParamsWithUse, newBody) = {
-              val newScope = outerScope.withImportReplacement(importReplacement)
-              val MethodDef(_, _, _, params, resultType, Some(body)) = getMethodBody(targetMethod)
-              transformIsolatedBody(Some(targetMethod), NoType, params, resultType, body)(newScope)
-            }
+            val newScope = outerScope.withImportReplacement(importReplacement)
 
-            if (!importReplacement.used.value.isUsed)
-              cancelFun()
+            val methodDef = getMethodBody(targetMethod)
+
+            transformCapturingBody(methodDef.args, targs, methodDef.body.get, OptEnv.Empty) {
+                (newCaptureParams, newCaptureValues, newBody) =>
+              if (!importReplacement.used.value.isUsed)
+                cancelFun()
```

Comment:
The indentation is the other way around here:
```suggestion
            transformCapturingBody(methodDef.args, targs, methodDef.body.get, OptEnv.Empty) {
              (newCaptureParams, newCaptureValues, newBody) =>
                if (!importReplacement.used.value.isUsed)
                  cancelFun()
```
and continuing in the following lines, of course.